test/python/circuit/test_library.py

```diff
@@ -144,34 +144,41 @@ def test_permutation_bad(self):
         self.assertRaises(CircuitError, Permutation, 4, [1, 0, -1, 2])
 
 
+@ddt
 class TestHiddenLinearFunctionLibrary(QiskitTestCase):
     """Test library of Hidden Linear Function circuits."""
+    def __q(self, x, hidden_function):
+        return np.dot(x.transpose(), np.dot(hidden_function, x))[0][0]
 
-    def test_hidden_linear_function(self):
-        """
-        Test hidden linear function circuit.
-        The following circuit is being tested for
-                ┌───┐   ┌───┐┌───┐
-        q_0: |0>┤ H ├─■─┤ S ├┤ H ├─────
-                ├───┤ │ └───┘├───┤
-        q_1: |0>┤ H ├─■───■──┤ H ├─────
-                ├───┤     │  ├───┤┌───┐
-        q_2: |0>┤ H ├─────■──┤ S ├┤ H ├
-                └───┘        └───┘└───┘
+    @data(
+        ([[1, 1, 0], [1, 0, 1], [0, 1, 1]],
+         HiddenLinearFunction([[1, 1, 0], [1, 0, 1], [0, 1, 1]]))
+    )
+    @unpack
```

Comment:
Could we use the same structure as for the other tests, e.g. the QFT: 
https://github.com/Qiskit/qiskit-terra/blob/076f852c5ea2165c591e6586c313b44b0cf57293/test/python/circuit/test_library.py#L305

That means the assertion functions are `camelCase` as the others (`assertAlmostEqual`, `assertTrue`, etc) but the actual test functions are `snake_case` (`test_something`):
```python
def assertHLFIsCorrect(A, hlf):
     # your implementation

@data(
    [[1, 1, 0], ...]  # just define A here
)
def test_hlf(self, A):
    hlf = HiddenLinearFunction(A)
    self.assertHLFIsCorrect(A, hlf)
```